mavros_extras/src/plugins/obstacle_distance.cpp

```diff
@@ -99,6 +109,44 @@ class ObstacleDistancePlugin : public plugin::PluginBase {
 
 		UAS_FCU(m_uas)->send_message_ignore_drop(obstacle);
 	}
+
+	void handle_obstacle_distance(const mavlink::mavlink_message_t *msg, mavlink::common::msg::OBSTACLE_DISTANCE &obstacle)
+	{
+		using mavlink::common::MAV_SENSOR_ORIENTATION;
+		using mavlink::common::MAV_DISTANCE_SENSOR;
+
+		auto laser_scan = boost::make_shared<sensor_msgs::LaserScan>();
+		// Not enough information for these
+		laser_scan->time_increment = 0.0;
+		laser_scan->scan_time = 0.0;
+		laser_scan->intensities = std::vector<float>();
+
+		laser_scan->header = m_uas->synchronized_header(frame_id, obstacle.time_usec);
+
+		laser_scan->range_min = obstacle.min_distance * 1E-2;	// Convert cm -> meters
+		laser_scan->range_max = obstacle.max_distance * 1E-2;
+
+		if (obstacle.increment_f > 0.0) {
+			laser_scan->angle_increment = obstacle.increment_f * DEG_TO_RAD;
+		} else {
+			laser_scan->angle_increment = obstacle.increment * DEG_TO_RAD;
+		}
+
+		// Transform FRD to ENU (90 degrees left, then flip axes)
+		laser_scan->angle_min = ((obstacle.angle_offset+270) % 360) * -DEG_TO_RAD;
+		laser_scan->angle_max = laser_scan->angle_min + laser_scan->angle_increment * obstacle.distances.size();
+		uint16_t dist;
+		for (int i=obstacle.distances.size(); i>0; i--) {	// TODO better memory map?
+			dist = obstacle.distances[i-1];
+			if (dist == UINT16_MAX){
+				laser_scan->ranges.push_back(INFINITY); // Discard all 'unknown/invalid' ranges
+			} else {
+				laser_scan->ranges.push_back(dist * 1E-2);
+			}
+		}
```

Comment:
I think something like that will be a bit better:
```cpp
laser_scan->ranges.reserve(obstacle.distances.size());
for (auto dist = obstacle.distabces.rbegin(); dist != obstacle.distances.rend(); dist++) {
    laser_scan->ranges.push_back((*dist == UINT16_MAX) ? INFINITY : *dist * 1E-2);
}

Reply:
It's also missing the transformation from local NED to local ENU.

Reply:
@mhkabir  We don't do any transforms on "send", so i don't think that we should transform anython on "receive" (and should remove transforms upper).

Reply:
Hmm, then that's wrong. We're kind of abusing the laserscan message in that case.